app/scripts/modules/core/src/pipeline/config/triggers/pubsub/pubsub.trigger.ts

```diff
@@ -0,0 +1,44 @@
+import {IController, module} from 'angular';
+
+import {SETTINGS} from 'core/config/settings';
+import {PIPELINE_CONFIG_PROVIDER} from 'core/pipeline/config/pipelineConfigProvider';
+import {SERVICE_ACCOUNT_SERVICE, ServiceAccountService} from 'core/serviceAccount/serviceAccount.service';
+import {IPubsubTrigger} from '@spinnaker/core';
+
+class PubsubTriggerController implements IController {
+
+  public fiatEnabled: boolean = SETTINGS.feature.fiatEnabled;
+  public pubsubTypes = ['kafka', 'google'];
+  public serviceAccounts: string[] = [];
+
+  constructor(public trigger: IPubsubTrigger, private serviceAccountService: ServiceAccountService) {
+    'ngInject';
+
+    this.serviceAccountService.getServiceAccounts().then(accounts => {
+      this.serviceAccounts = accounts || [];
+    });
+  }
+}
+
+export const PUBSUB_TRIGGER = 'spinnaker.core.pipeline.trigger.pubsub';
+module(PUBSUB_TRIGGER, [
+  SERVICE_ACCOUNT_SERVICE,
+  PIPELINE_CONFIG_PROVIDER,
+]).config((pipelineConfigProvider: any) => {
+  pipelineConfigProvider.registerTrigger({
```

Comment:
I think this addresses your comment below - I would feature flag thisl:

```
if (SETTINGS.feature.pubsubTriggerEnabled) {
  pipelineConfigProvider.registerTrigger({ ... });
}
```

Not sure if Deck will break if `pubsub` is included in the default triggers but not registered.